types/react-native/index.d.ts

```diff
@@ -3491,7 +3491,7 @@ export interface FlatListProperties<ItemT> extends VirtualizedListProperties<Ite
     /**
      * Rendered in between each item, but not at the top or bottom
      */
-    ItemSeparatorComponent?: React.ComponentClass<any> | null
+    ItemSeparatorComponent?: React.ComponentClass<any> | React.ReactElement<any> | (() => React.ReactElement<any>) | null
```

Comment:
Oh, I understood that the idea was to keep the React.ComponentClass and then the other two for `ReactElement`, sorry about that misunderstanding.

From what I've tested while getting the PR ready, `React.ComponentClass` does not work for  `React.StatelesComponents`, `ItemSeparatorComponent?: React.ComponentClass<any> | null` was the one that the spec had before I opened the issue. 

I'm not entirely sure about the `ReactElement<any>` if that should not be permitted then the solution that works is `ItemSeparatorComponent?: React.ComponentClass<any> | React.StatelessComponent<any> | null`

Again, sorry about that misunderstanding.